sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ParallelDocumentQueryExecutionContext.java

```diff
@@ -106,7 +106,8 @@ public static <T> Flux<IDocumentQueryExecutionComponent<T>> createAsync(
                         || queryInfo.hasAggregates()
                         || queryInfo.hasGroupBy()
                         || queryInfo.hasDCount()
-                        || queryInfo.hasDistinct()),
+                        || queryInfo.hasDistinct()
+                        || queryInfo.hasNonStreamingOrderBy()),
```

Comment:
Please also add the NonStreamingOrderBy in `canCacheQuery`

Reply:
Added